closed/custom/Docs.gmk

```diff
@@ -116,3 +116,174 @@ $(OPENJ9_JAVADOC_MARKER) : $(OPENJDK_JAVADOC_MARKER)
+JDK_API_CUSTOM_TARGETS += $(OPENJ9_ONLY_JAVADOC_MARKER)
+
+OPENJ9_ONLY_JDK_NAME := OpenJ9 JDK $(VERSION_SPECIFICATION)
+OPENJ9_ONLY_DOCDIR := $(IMAGES_OUTPUTDIR)/openj9-docs/api
+OPENJ9_ONLY_INDEX_HTML := $(IMAGES_OUTPUTDIR)/openj9-docs/api/index.html
+
+# Create a list of the openj9 source files for which javadoc is required.
+# These are the openj9.xxxx packages and the jdk management packages with
+# public methods which are openj9 specific extensions or which modify the
+# openjdk implementations behaviour.  javadoc is not generated for the
+# classes  which are replaced by openj9 but are functionally identical
+# to the openjdk implmentations (e.g. java.lang.String), since the openjdk
+# javadoc is still applicable.
+# 
+# The list if source files is passed explicitly (via OPENJ9_ONLY_FILES)
+# to the javadoc tool on the command line and used as dependencies
+# (via OPENJ9_ONLY_FILELIST) to identify when the javadoc needs regenerating.
+
+OPENJ9_ONLY_FILES := $(SUPPORT_OUTPUTDIR)/openj9-docs/_javadoc_openj9_files.txt
+CREATE_FILELIST_RESULT := $(shell $(MKDIR) -p $(SUPPORT_OUTPUTDIR)/openj9-docs ; $(FIND) $(SUPPORT_OUTPUTDIR)/j9jcl_sources -type f -path '*/openj9.*/*.java' -o -path '*/jdk.management/*.java' >$(OPENJ9_ONLY_FILES))
+ifneq ($(CREATE_FILELIST_RESULT), )
+  $(error closed/custom/Docs.gmk: Error creating $(OPENJ9_ONLY_FILES): $(CREATE_FILELIST_RESULT))
+endif
+OPENJ9_ONLY_FILELIST := $(shell $(CAT) $(OPENJ9_ONLY_FILES))
+
+# Set up the javadoc command line options.
+# Use  -linkoffline to redirect the links to the documentation for the classes
+# from the openjdk project source files to the# online Oracle documentation.
+# See the  javadoc documentation for details of the linkoffline option - e.g.
+# https://docs.oracle.com/en/java/javase/14/javadoc/javadoc.html
+# Deferred expansion is required for variables because this file is included
+# by make/Docs.gmk before variables such as JAVADOC_OPTIONS and JAVADOC_TAGS
+# have been defined.
+OPENJ9_ONLY_JAVADOC_OPTIONS = $(filter-out --expand-requires transitive,$(JAVADOC_OPTIONS))
+OPENJ9_ONLY_JAVADOC_OPTIONS += -linkoffline $(JAVASE_BASE_URL)/ $(DOCS_OUTPUTDIR)/api/
+OPENJ9_ONLY_JAVADOC_OPTIONS += $(JAVADOC_TAGS)
+OPENJ9_ONLY_JAVADOC_OPTIONS += --module-source-path $(OPENJ9_STUBS_DIR)$(PATH_SEP)$(MODULES_SOURCE_PATH)
+OPENJ9_ONLY_JAVADOC_OPTIONS += -Xdoclint:all,$(call CommaList, $(addprefix -, $(JAVADOC_DISABLED_DOCLINT)))
+
+OPENJ9_DOC_TITLE = $(JDK_LONG_NAME)<br>Version $(VERSION_SPECIFICATION)
+OPENJ9_WINDOW_TITLE = $(subst &amp;,&,$(JDK_SHORT_NAME))
+OPENJ9_HEADER_TITLE = <div $(HEADER_STYLE)><strong>$(JDK_SHORT_NAME)</strong></div>
+
+OPENJ9_ONLY_JAVADOC_OPTIONS += -doctitle '$(OPENJ9_ONLY_JDK_NAME)'
+OPENJ9_ONLY_JAVADOC_OPTIONS += -windowtitle '$(OPENJ9_ONLY_JDK_NAME)'
+OPENJ9_ONLY_JAVADOC_OPTIONS += -header '<div $(HEADER_STYLE)><strong>$(OPENJ9_ONLY_JDK_NAME)</strong></div>'
+OPENJ9_ONLY_JAVADOC_OPTIONS += -bottom '$(OPENJ9_JAVADOC_BOTTOM)'
```

Comment:
The line above:
```
OPENJ9_JAVADOC_BOTTOM := $(subst &,\&,$(OPENJ9_JAVADOC_BOTTOM))
```
should be removed and the use of `$(subst)` inlined into `OPENJ9_SUBST_COMMAND` to avoid the extra `\`.

Reply:
Fixed